.github/workflows/ci.yml

```diff
@@ -9,36 +9,29 @@ on:
 
 jobs:
   check:
-    name: check
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        build: [msrv, stable]
         features:
           [
             "",
             "--no-default-features",
             "--no-default-features --features use_alloc",
             "--all-targets --all-features",
           ]
-        include:
-          - build: msrv
-            rust: 1.62.1
-          - build: stable
-            rust: stable
-        exclude:
-          - build: msrv
-            # we only care about the MSRV with respect to the lib target
-            features: "--all-targets --all-features"
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ matrix.rust }}
+      - uses: dtolnay/rust-toolchain@stable
       - run: cargo check ${{ matrix.features }}
 
+  msrv:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: taiki-e/install-action@cargo-hack
```

Comment:
Thanks! I didn't know `cargo hack` could do this! 

Just add the `actions/checkout` dependency before this, and I think CI will pass.